plugins/backend/catalog-backend-module-okta/src/providers/OktaGroupEntityProvider.ts

```diff
@@ -136,14 +143,28 @@ export class OktaGroupEntityProvider extends OktaEntityProvider {
           group,
           oktaGroups,
         });
-
+        const profileAnnotations: Record<string, string> = {};
+        if (this.customAttributesToAnnotationAllowlist.length) {
+          for (const [key, value] of new Map(Object.entries(group.profile))) {
+            const stringKey = key.toString();
+            if (
+              this.customAttributesToAnnotationAllowlist.includes(stringKey)
+            ) {
+              profileAnnotations[stringKey] = value.toString();
```

Comment:
I think how we serialize this value could be better.. It can be any of these types:

```boolean | number | string | boolean[] | number[] | string[]```

Maybe we could change this to:

```
profileAnnotations[stringKey] = JSON.stringify(value);
```

Reply:
Here is some examples. See the true value as a string is undecernable from the the boolean true using toString, but it is correct using JSON.stringify.

```
> true.toString()
'true'
> "true".toString()
'true'
> JSON.stringify("true")
'"true"'
> JSON.stringify(true)
'true'
```